### PR TITLE
fix: microsoft clarity load and handle promise for identify

### PR DESCRIPTION
**Pull request by sanpj2292**
## PR Description

Microsoft Clarity destination in Rudderstack only supports `identify` event & when we try to send `identify` event to Clarity would return a Promise
But while SDK is upgrading/loading, seems like the current logic of validating if the SDK has been loaded is not complete & caused unhandled exceptions([here](https://app.bugsnag.com/rudderlabs/sdk-error-reporter-js/errors/6756a0437c1a664cc5580beb?i=wh&m=nw&pivot_tab=event))

Through this PR, we are planning to fix the issue

## Linear task (optional)

Resolves INT-2937

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Improved loading checks for the Microsoft Clarity integration.
	- Enhanced error handling during the user identification process.

- **Bug Fixes**
	- Ensured traits are set only after successful identification.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js [hunk 1]
```diff
@@ -55,7 +56,11 @@ class MicrosoftClarity {
     if (context?.traits?.customPageId) {
       customPageId = context.traits.customPageId;
     }
-    window.clarity('identify', userId, sessionId, customPageId);
+
+    const identifyPromise = window.clarity('identify', userId, sessionId, customPageId);
+    if (!!identifyPromise) { // Clarity SDK is ready
+      identifyPromise.catch(logger.error);
+    }
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:63]: The implemented version is similar to the suggestion. Can you check and confirm ?

[the remark this answers is not in the document]
Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:63]: We don't need to log the success scenario

**Review by koladilip — DISMISSED** [review]

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js [hunk 2]
```diff
@@ -55,7 +56,11 @@ class MicrosoftClarity {
     if (context?.traits?.customPageId) {
       customPageId = context.traits.customPageId;
     }
-    window.clarity('identify', userId, sessionId, customPageId);
+
+    const identifyPromise = window.clarity('identify', userId, sessionId, customPageId);
+    if (!!identifyPromise) { // Clarity SDK is ready
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:61]: To be more cautious, let's check for the promise type here.
```
if (typeof identifyPromise?.then === 'function')

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js [hunk 3]
```diff
@@ -55,7 +56,11 @@ class MicrosoftClarity {
...
+      identifyPromise.catch(logger.error);
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:62]: Let's be more explicit in the log message similar to the first version of these changes.

```
identifyPromise.catch(error => {
      logger.error('The "identify" promise was rejected', error);
});
```

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js (hunk 2 above)
Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js]: Done

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js (hunk 3 above)
Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js]: Done

**Review by saikumarrs — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js [hunk 4]
```diff
@@ -55,7 +56,13 @@ class MicrosoftClarity {
     if (context?.traits?.customPageId) {
       customPageId = context.traits.customPageId;
     }
-    window.clarity('identify', userId, sessionId, customPageId);
+
+    const identifyPromise = window.clarity('identify', userId, sessionId, customPageId);
+    if (typeof identifyPromise?.then === 'function') { // Clarity SDK is ready
```
Comment (koladilip) [inline packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:61]: this is fine but just a suggestion:
https://stackoverflow.com/a/27760489/18047180

**Review by koladilip — APPROVED** [review]

File: packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js (hunk 4 above)
Comment (koladilip) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:61]: Feel free to ignore

Comment (sanpj2292) [reply packages/analytics-js-integrations/src/integrations/MicrosoftClarity/browser.js:61]: Ignoring for now

**Pull request merged**